src/upnp_xml.cc

```diff
@@ -841,10 +845,12 @@ void UpnpXMLBuilder::addResources(const std::shared_ptr<CdsItem>& item, pugi::xm
             continue;
         }
 
+        // TODO FIXME Should these be attributes or something else?
```

Comment:
The goal in this method is to calculate temporary properties depending on the client and return them. We should not make them permanent, e.g. the raw data for _protocolInfo_ is in attributes, extended by more details and returned to the client.

We may add another attribute set `std::map<std::string, std::string> responseAttributes` to the resource for this purpose allowing more sophisticated handling and storage along with the request.

I recommend doing this separately.

Reply:
`clientSpecficAttrs` added in latest rev